devices/power.py

```diff
@@ -299,6 +312,123 @@ def reset(self):
+        Snmp("set",self.ip_address,"private",oid ,'i',0)
+        result = Snmp("get",self.ip_address,"private",oid)
+        print("result = %s" % result)
+        return result
+    
+    def reset(self):
+	self.off()
+	time.sleep(5)
+	self.on()
+    
+    def getPowerStatus(self):
+        oid = self.oid_Outlet + '.' + str(self.port) + '.0'
+        result = Snmp("get",self.wan_ip,"private",oid)
+        return result
+    
+    def getPowerStatusAll(self):
+        oid = self.oid_Outlet
+        result = Snmp("walk",self.wan_ip,"private",oid)
+        return result
+
+def Snmp(action=None, target_IP=None, community='public', oid=None, stype=None, value=None, walk_timeout=120):
+    """
+    Name:Snmp
+    Purpose: Do snmp set,get and walk function for DUT control.
+    Input:1.action
+    	    2.target_IP
+         3.community
+         4.oid
+         5.stype
+         6.value
+         7.walk_timeout
+    Output:Snmp get or walk result
+    """
+
+    # Create command generator
+    cmdGen = cmdgen.CommandGenerator()
+
+    if action == "get":
+        errorIndication, errorStatus, errorIndex, varBinds = cmdGen.getCmd(
+            cmdgen.CommunityData(community),
+            cmdgen.UdpTransportTarget((target_IP, 161), timeout=1, retries=3),
+            oid
+        )
+        for name, val in varBinds:
+            if val.prettyPrint() != "":
+                return val.prettyPrint()
+            else:
+                return "null"
+    elif action == "set":
+        for case in switch(stype):
+            if case('i'): 
+                value = rfc1902.Integer(value)
+                break
+            elif case('s'): 
+                value = rfc1902.OctetString(value)
+                break 
+            elif case('x'): 
+                value = rfc1902.OctetString(hexValue=value)
+                break 
+            elif case('p'): 
+                value = rfc1902.IpAddress(value)
+            elif case('u'): 
+                value = rfc1902.Unsigned32(value)
+                break 
+            elif case(): # default
+                value = ""
+        
+        cmdGen.setCmd(
+            cmdgen.CommunityData(community),
+            cmdgen.UdpTransportTarget((target_IP, 161), timeout=1, retries=3),
+            (oid, value)
+        )
+
+class switch(object):
+    """
+    Name:switch
+    Purpose: def switch/case library for Snmp set used.
+    Input:object ex.stype
+    Output:True or False
+    """
+    
+    def __init__(self, value):
+        self.value = value
+        self.fall = False
+ 
+    def __iter__(self):
+        """Return the match method once, then stop"""
+        yield self.match
+     
+    def match(self, *args):
+        """Indicate whether or not to enter a case suite"""
+        if self.fall or not args:
+            return True
+        elif self.value in args: 
+            self.fall = True
+            return True
+        else:
+            return False
+            
+
```

Comment:
{
           "stationname": {
                "board_type": "mv1",
                "conn_cmd": ["ssh pi@192.168.100.21 cu -l /dev/ttyUSB1 -s 115200", "ssh pi@192.168.100.21 cu -l /dev/ttyUSB0 -s 115200" ],
                "location": "unknown",
		"powerport": "cmd://on_cmd=snmpset -v 2c -c 7oWm4MC IPADDRESS .1.3.6.1.4.1.39525.11.1.2.5 i 1;off_cmd=snmpset -v 2c -c 7oWm4MC IPADDRESS .1.3.6.1.4.1.39525.11.1.2.5 i 0",
		"devices": [
			{
				"type": "debian",
				"name": "lan",
				"ipaddr": "192.168.100.21",
				"color": "blue",
				"username": "pi",
				"options": "tftpd-server"
			}
		],
		"connection_type": "local_cmd"
        }
}

Reply:
We will modify to support http to control power and modify pull request.